clients/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -97,6 +121,19 @@ public Collection<PubSubMessage<K, ChangeEvent<V>, Long>> poll(long timeoutInMs)
   // TODO: Find a better way to avoid data gap between version topic and change capture topic due to log compaction.
   private boolean handleControlMessage(ControlMessage controlMessage, PubSubTopicPartition pubSubTopicPartition) {
     ControlMessageType controlMessageType = ControlMessageType.valueOf(controlMessage);
+    if (controlMessageType.equals(ControlMessageType.START_OF_PUSH)) {
+      StartOfPush startOfPush = (StartOfPush) controlMessage.controlMessageUnion;
+      byte[] dictionary = null;
+      if (startOfPush.compressionDictionary != null) {
+        dictionary = startOfPush.compressionDictionary.array();
+      }
+      // TODO: This relies on consuming the beginning of the version topic. This is what some libraries do anyway under
```

Comment:
Based on my understanding of this AfterImageConsumerClient it will always ingest version topic and then switch to CC topic, so this is actually expected right? curious what do you want to achieve at the end state?

Reply:
Good question.  The reason why this isn't great is because we have clients who will want to seek to an after doing some work.  This means that all users of this interface if they use a 'seek' api aren't guaranteed to hit the start of push message.

We don't have the seek function in the api yet, so this is fine for now.  But we'll need to tweak this soon.